pkg/credentials/credentials.go

```diff
@@ -0,0 +1,83 @@
+/*
+Copyright 2021 the Velero contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package credentials
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+type Getter interface {
+	Get(selector *corev1.SecretKeySelector) ([]byte, error)
+
+	// GetAsFile gets the secret data specified by the selector and returns a filepath
+	// where the contents have been written to
+	GetAsFile(selector *corev1.SecretKeySelector) (string, error)
+}
+
+type credentialsGetter struct {
```

Comment:
Per community discussion, this may need a field for the name of the env variable that it will land in.